.circleci/config.yml

```diff
@@ -36,34 +19,55 @@ jobs:
             - dependency-cache-{{ checksum "package-lock.json" }}
 
       - run:
-          name: Install cqlsh
-          # We need to edit DEFAULT_CQLVER defined in the cqlsh script
-          # to connect to the server without --cqlversion command line option
+          name: Docker login
           command: |
-            curl -O https://bootstrap.pypa.io/get-pip.py
-            sudo python get-pip.py
-            sudo pip install cqlsh
-            sudo sed -i "s/^DEFAULT_CQLVER = .*/DEFAULT_CQLVER = '3.4.4'/" /usr/local/bin/cqlsh
+            echo ${GHCR_PAT} | docker login https://ghcr.io -u ${GHCR_USERNAME} --password-stdin
 
       - run:
-          name: Wait for Cassandra
-          # dockerize command is included in circleci/openjdk:8-jdk image
-          command: dockerize -wait tcp://127.0.0.1:9042 -timeout 1m
+          name: Update docker and docker compose
+          command: |
+            sudo apt-get purge docker-ce containerd.io
+            curl -fsSL https://get.docker.com -o get-docker.sh
+            sudo sh get-docker.sh
+            sudo curl -L "https://github.com/docker/compose/releases/download/1.26.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+            sudo chmod +x /usr/local/bin/docker-compose
 
       - run:
-          name: Install schema on Cassandra
-          command: cqlsh --execute "$SCHEMA_CONTENT" 127.0.0.1 9042
+          name: Run docker-compose up
+          command: docker-compose up -d
+          working_directory: .circleci/
 
       - run:
-          name: Install schema for function purpose on Cassandra
-          command: cqlsh -f .circleci/function_schema.cql 127.0.0.1 9042
+          name: Install node@14.15.1 # Update node for karma to work
+          command: |
+            export NVM_DIR="/opt/circleci/.nvm"
+            [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+            nvm install v14.15.1
+            nvm alias default v14.15.1
+            echo 'export NVM_DIR="/opt/circleci/.nvm"' >> $BASH_ENV
+            echo "[ -s \"$NVM_DIR/nvm.sh\" ] && . \"$NVM_DIR/nvm.sh\"" >> $BASH_ENV
 
       - run:
-          name: Wait for Scalardl
-          command: dockerize -wait tcp://127.0.0.1:50051 -timeout 1m
+          name: Install npm
+          command: npm install
 
       - run:
-          command: npm install
+          name: Wait for cassandra to spin up
+          command: |
+            echo -n "Wait for C* "
+              until docker exec circleci_scalardl-cassandra-1 cqlsh 2> /dev/null ; do
+                  sleep 0.5
+                  echo -n "#"
+              done
+
+      - run:
+          name: Insert function table and wait
+          command: |
+            docker cp function_schema.cql \
+            circleci_scalardl-cassandra-1:function_schema.cql
+            docker exec circleci_scalardl-cassandra-1 cqlsh -f function_schema.cql
+            sleep 10
```

Comment:
Sorry i couldn't think of a good wait to wait for the container to be ready.

Reply:
Without the sleep, the test fails I guess ?

Reply:
Yes, i will fail due to the container being able to be ready on time.

Reply:
Waiting what container to be ready?